otx/mpa/modules/models/heads/semisl_cls_head.py

```diff
@@ -6,47 +6,24 @@
 from mmcls.models.builder import HEADS
 from mmcls.models.heads.linear_head import LinearClsHead
 
+from otx.mpa.modules.models.heads.non_linear_cls_head import NonLinearClsHead
 
-@HEADS.register_module()
-class SemiSLClsHead(LinearClsHead):
-    """Semi-SL for Classification Head
 
-    This ClsHead is a classification linear head based on the FixMatch algorithm.
-    This head uses the dynamic threshold for each class calculated based on
-    the confidence value of the model.
+class SemiClsHead:
```

Comment:
I'm very sorry but I still can't understand your refactored code as well. I think it would better inherit both LinearClsHead and NonLinearClsHead here and remove all SemiLinearClsHead or SemiNonLinearClsHead, remaining only this class. If that's impossible, it's okay to leave it as it. Plus, I think SemiSLClsHead is more proper name.